From 360d7eb0f8233d16557ef34a9e58055a67ea9b70 Mon Sep 17 00:00:00 2001
From: Michal Sekletar <msekleta@redhat.com>
Date: Mon, 12 Jun 2017 13:43:48 +0200
Subject: [PATCH] rules: move cpu hotplug rule to separate file

In kdump initrd we don't want to automatically bring online all
available CPUs. Hence, kdump maintainers can easily mask the rule by
placing symlink to /dev/null with the same name in /etc/udev/rules.d

RHEL-only

Related: #1266322

[msekleta: note that this is just part of the fix for #1266322, in
order to fix the bug it is necessary to actually mask the newly added
rule when generating kdump initrd]
---
 Makefile.am                       | 1 +
 rules/40-redhat-cpu-hotplug.rules | 4 ++++
 rules/40-redhat.rules             | 3 ---
 3 files changed, 5 insertions(+), 3 deletions(-)
 create mode 100644 rules/40-redhat-cpu-hotplug.rules

diff --git a/Makefile.am b/Makefile.am
index e9ceac98a..94fee02c5 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -3511,6 +3511,7 @@ dist_udevrules_DATA += \
 	rules/80-net-setup-link.rules \
 	rules/95-udev-late.rules \
 	rules/40-redhat.rules \
+	rules/40-redhat-cpu-hotplug.rules \
 	rules/73-idrac.rules \
         rules/80-net-name-slot.rules
 
diff --git a/rules/40-redhat-cpu-hotplug.rules b/rules/40-redhat-cpu-hotplug.rules
new file mode 100644
index 000000000..bc5ddc841
--- /dev/null
+++ b/rules/40-redhat-cpu-hotplug.rules
@@ -0,0 +1,4 @@
+# do not edit this file, it will be overwritten on update
+
+# CPU hotadd request
+SUBSYSTEM=="cpu", ACTION=="add", TEST=="online", ATTR{online}=="0", ATTR{online}="1"
diff --git a/rules/40-redhat.rules b/rules/40-redhat.rules
index 34a1df9c4..d04c7fc9a 100644
--- a/rules/40-redhat.rules
+++ b/rules/40-redhat.rules
@@ -1,8 +1,5 @@
 # do not edit this file, it will be overwritten on update
 
-# CPU hotadd request
-SUBSYSTEM=="cpu", ACTION=="add", TEST=="online", ATTR{online}=="0", ATTR{online}="1"
-
 # Memory hotadd request
 SUBSYSTEM=="memory", ACTION=="add", PROGRAM="/bin/uname -p", RESULT!="s390*", ATTR{state}=="offline", ATTR{state}="online"
 
